python/pyspark/ml/classification.py

```diff
@@ -134,7 +137,7 @@ class Classifier(Predictor, _ClassifierParams, metaclass=ABCMeta):
     """
 
     @since("3.0.0")
-    def setRawPredictionCol(self, value):
+    def setRawPredictionCol(self, value: str) -> "P":
```

Comment:
All the setter methods in classes intended for inheritance should have bounds on `self` to ensure that specific type is resolved on return, i.e.

```python
def setRawPredictionCol(self: "P", value: str) -> "P":
```